mlir/lib/Quantum/Utils/QuantumSplitting.cpp

```diff
@@ -89,17 +91,73 @@ void AugmentedCircuitGenerator::generate(Region &region, OpBuilder &builder)
         else if (auto extractOp = dyn_cast<quantum::ExtractOp>(op)) {
             cacheDynamicWire(extractOp, builder);
         }
-        else if (auto gate = dyn_cast<quantum::DifferentiableGate>(op)) {
-            ValueRange diffParams = gate.getDiffParams();
-            if (!diffParams.empty()) {
-                for (Value param : diffParams) {
-                    builder.create<ListPushOp>(gate.getLoc(), oldToCloned.lookupOrDefault(param),
-                                               cache.paramVector);
+        else if (auto gate = dyn_cast<quantum::ParametrizedGate>(op)) {
+            ValueRange params = gate.getAllParams();
+
+            for (Value param : params) {
+                Location loc = gate.getLoc();
+                Value clonedParam = oldToCloned.lookupOrDefault(param);
+                Type paramType = clonedParam.getType();
+
+                if (!paramType.dyn_cast<RankedTensorType>()) {
+                    builder.create<ListPushOp>(loc, clonedParam, cache.paramVector);
```

Comment:
I think this branch doesn't verify that the datatype is `f64`.

Reply:
We only have parameters `f64` or `tensor<complex<f64>>`. I'll change it to `f64`.

Reply:
Done.